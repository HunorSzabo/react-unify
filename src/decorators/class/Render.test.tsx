import * as React from 'react';
import { create } from 'react-test-renderer';
import { prop, Render, state } from '../';

@Render((counter: Counter) => (
  <div>
    <p>Count: {counter.count}</p>
    <button onClick={() => counter.increment()}>
      Increment by {counter.amount}
    </button>
  </div>
))
class Counter extends React.Component<{
  amount?: number;
  newCount?: number;
}> {
  static Render: (counter: Partial<Counter>) => any;

  static getDerivedStateFromProps(nextProps) {
    const newCount = nextProps.newCount;
    return newCount == null ? null : { count: newCount };
  }

  @prop amount: number = 1;
  @prop newCount!: number;
  @state count: number = 0;

  increment() {
    this.count += this.amount;
  }
}

describe('@Render', () => {
  it('renders Counter component', () => {
    const renderer = create(<Counter />);
    renderer.root.findByType(Counter).instance.increment();
    renderer.root.findByType('button').props.onClick();
    expect(renderer).toMatchSnapshot();
  });

  it('updates from getDerivedStateFromProps', () => {
    const renderer = create(<Counter />);
    renderer.update(<Counter newCount={1} />);
    expect(renderer).toMatchSnapshot();
  });

  it('sets Counter.Render', () => {
    expect(Counter.Render({ count: 0, amount: 1 })).toMatchSnapshot();
  });
});
