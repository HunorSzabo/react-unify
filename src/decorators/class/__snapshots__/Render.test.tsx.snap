// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Render renders Counter component 1`] = `
<div>
  <p>
    Count: 
    2
  </p>
  <button
    onClick={[Function]}
  >
    Increment by 
    1
  </button>
</div>
`;

exports[`@Render sets Counter.Render 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment by 
    1
  </button>
</div>
`;

exports[`@Render updates from getDerivedStateFromProps 1`] = `
<div>
  <p>
    Count: 
    1
  </p>
  <button
    onClick={[Function]}
  >
    Increment by 
    1
  </button>
</div>
`;
